x/profiles/module_ibc.go

```diff
@@ -117,29 +119,62 @@ func (am AppModule) OnChanCloseConfirm(
 // OnRecvPacket implements the IBCModule interface
 func (am AppModule) OnRecvPacket(
 	ctx sdk.Context,
-	modulePacket channeltypes.Packet,
+	packet channeltypes.Packet,
 ) (*sdk.Result, []byte, error) {
+	var ack channeltypes.Acknowledgement
+	var err error
+
+	// Try handling the chain link packet data
+	ack, err = am.handleLinkChainAccountPacketData(ctx, packet)
+	if types.ErrInvalidPacketData.Is(err) {
+		// Try handling the oracle request packet data
+		ack, err = am.handleOracleRequestPacketData(ctx, packet)
+	}
+
+	// If packet data is still invalid, return an error
+	if types.ErrInvalidPacketData.Is(err) {
+		return nil, nil, err
+	}
```

Comment:
Cant it be switch case to handle packet data?

Say, 
```
switch packet.(type) {
case *types.LinkChainAccountPacketData:
  am.handleLinkChainAccountPacketData(ctx, packet)
case *oracletypes.OracleResponsePacketData:
  am.handleOracleRequestPacketData(ctx, packet)
default:
  return nil, nil, err
}
```

Reply:
Unfortunately, this is not possible since `Packet` is not an interface

Reply:
@RiccardoM I see. It makes sense to me now.